miner/worker.go

```diff
@@ -834,14 +843,19 @@ func (miner *Miner) validateParams(genParams *generateParams) (time.Duration, er
 	return time.Duration(blockTime) * time.Second, nil
 }
 
-func (miner *Miner) blockFeeCurrency(env *environment, feeCurrency common.Address, err error) {
+func (miner *Miner) registerFeeCurrencyTxFailure(env *environment, tx *types.Transaction, err error) {
 	// the fee-currency is still in the allowlist of this environment,
 	// so set the fee-currency block gas limit to 0 to prevent other
 	// transactions.
-	pool := env.multiGasPool.PoolFor(&feeCurrency)
-	pool.SetGas(0)
+	if env.blockingAllowed {
+		pool := env.multiGasPool.PoolFor(tx.FeeCurrency())
+		pool.SetGas(0)
+	}
 	// also add the fee-currency to a worker-wide blocklist,
 	// so that they are not allowlisted in the following blocks
 	// (only locally in the txpool, not consensus-critical)
-	miner.feeCurrencyBlocklist.Add(feeCurrency, *env.header)
+	miner.feeCurrencyBlocklist.Add(*tx.FeeCurrency(), *env.header)
```

Comment:
we're sure that tx.FeeCurrency is never nil here?

Reply:
Previously this function was being called with the de-referenced fee currency [here](https://github.com/celo-org/op-geth/pull/345/files#diff-689874b30f6f905ce6c47cdefeddcf052b467a7936a981f4b6cf38939e10d924L427) so I'm as sure as we previously were that fee currency is not nil.